### PR TITLE
[BFD 714] Adds a reader node to the aurora db cluster

### PR DESCRIPTION
**Pull request by keithdadkins**
### Change Details
This PR adds a reader node to our db cluster in preparation for Mitre's Synthia work. Ultimately, this reader node will be assigned a custom endpoint that the Synthia team can use without impacting prod.

This only adds a reader node to the cluster and does not include custom endpoints.

### Acceptance Validation
- [x] Terraform planned and reviewed to ensure this action will **ONLY** add a reader node.
- [x] The cost of the extra reader is discussed and approved with CMS
- [x]  Adding a reader node may cause an availability impact for a minute or two, so care should be take when this is applied (e.g., apply off peak hours). Note: Our terraform will 'apply immediately' and not during a maintenance window.

### Feedback Requested
No glaring issues.

### External References

- [BFD-714](https://jira.cms.gov/browse/BFD-714)

### Security Implications

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Comment by keithdadkins** [comment]
Prelim thumbs up from CMS regarding costs for the extra reader at least until July.

File: ops/terraform/env/prod/stateful/main.tf [hunk 1]
```diff
@@ -12,7 +12,10 @@ module "stateful" {
 
   aurora_config = {
     instance_class = "db.r5.12xlarge"
-    cluster_nodes  = 3
+
+    # With aurora you do not designate primary and replicas. Instead, you simply add RDS Instances and
```
Comment (jzulim) [inline ops/terraform/env/prod/stateful/main.tf:16]: 👍

File: ops/terraform/modules/resources/aurora/main.tf [hunk 2]
```diff
@@ -1,7 +1,9 @@
 locals {
+  # note: this captures both prod and prod-sbx
```
Comment (jzulim) [inline ops/terraform/modules/resources/aurora/main.tf:2]: Helpful comment, thx

File: ops/terraform/modules/resources/aurora/main.tf [hunk 3]
```diff
@@ -101,7 +103,8 @@ resource "aws_rds_cluster_instance" "aurora_nodes" {
   identifier         = "${lookup(local.node_identifier, var.env_config.env)}-node-${count.index}"
   cluster_identifier = aws_rds_cluster.aurora_cluster.id
 
-  availability_zone       = var.stateful_config.azs[count.index]
+  # note: element(list,index) function wraps. e.g., element([az1,az2,az3],3) == az1
```
Comment (jzulim) [inline ops/terraform/modules/resources/aurora/main.tf:106]: fancy. :)

**Review by jzulim — APPROVED** [review]
LGTM. Any chance we can get a terraform plan for SBX and PROD. Will this add a reader node for SBX as well (I believe it will) - If true, any way to not add a reader for SBX without a ton of TF refactoring? If not, and we want to use this reader node in SBX for testing purposes, I'm cool with that but perhaps we need to surface this cost if not already? Whichever the outcome, this LGTM. Liking the additional comments.

**Review by Dranith — APPROVED** [review]
Not a terraform expert, but this looks pretty clear.

**Comment by keithdadkins** [comment]
> LGTM. Any chance we can get a terraform plan for SBX and PROD. Will this add a reader node for SBX as well (I believe it will) - If true, any way to not add a reader for SBX without a ton of TF refactoring? If not, and we want to use this reader node in SBX for testing purposes, I'm cool with that but perhaps we need to surface this cost if not already? Whichever the outcome, this LGTM. Liking the additional comments.

As soon as AWS gets unscrewed I will run a plan and drop it in slack. For now, I cannot query the API due to the ongoing outage. But, it should **only** add a reader node to prod.

**Review by rtravitz — APPROVED** [review]
:shipit:

**Pull request merged**